sessionctx/variable/sysvar.go

```diff
@@ -1953,6 +1953,23 @@ var defaultSysVars = []*SysVar{
 		s.OptPrefixIndexSingleScan = TiDBOptOn(val)
 		return nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBExternalTS, Value: strconv.FormatInt(DefTiDBExternalTS, 10), SetGlobal: func(ctx context.Context, s *SessionVars, val string) error {
+		ts, err := parseTSFromNumberOrTime(s, val)
+		if err != nil {
+			return err
+		}
+		return SetExternalTimestamp(ctx, ts)
+	}, GetGlobal: func(ctx context.Context, s *SessionVars) (string, error) {
+		ts, err := GetExternalTimestamp(ctx)
+		if err != nil {
+			return "", err
+		}
+		return strconv.Itoa(int(ts)), err
+	}},
+	{Scope: ScopeGlobal | ScopeSession, Name: TiDBEnableExternalTSRead, Value: BoolToOnOff(false), Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
```

Comment:
`tidb_enable_external_ts_read` has a global scope. I have a concern that whether it will affect some internal sqls which should not use external ts.

Reply:
Should I verify whether it's from internal or normal user connection, and skip external ts staleness for all internal execution? I think it's more robust.

It seems to have caused some problem during the test (about setting system variables...), but I'm not sure and cannot reproduce now.

Reply:
Added several conditions to skip all staleness logic for internal usage.